nixos/modules/services/networking/prosody.nix

```
@@ -755,105 +840,11 @@ in
-      };
-
-      allow_registration = ${toLua cfg.allowRegistration}
-
-      c2s_require_encryption = ${toLua cfg.c2sRequireEncryption}
-
-      s2s_require_encryption = ${toLua cfg.s2sRequireEncryption}
-
-      s2s_secure_auth = ${toLua cfg.s2sSecureAuth}
-
-      s2s_insecure_domains = ${toLua cfg.s2sInsecureDomains}
-
-      s2s_secure_domains = ${toLua cfg.s2sSecureDomains}
-
-      authentication = ${toLua cfg.authentication}
-
-      http_interfaces = ${toLua cfg.httpInterfaces}
-
-      https_interfaces = ${toLua cfg.httpsInterfaces}
-
-      http_ports = ${toLua cfg.httpPorts}
-
-      https_ports = ${toLua cfg.httpsPorts}
-
-      ${ cfg.extraConfig }
-
-      ${lib.concatMapStrings (muc: ''
-        Component ${toLua muc.domain} "muc"
-            modules_enabled = { "muc_mam"; ${optionalString muc.vcard_muc ''"vcard_muc";'' } }
-            name = ${toLua muc.name}
-            restrict_room_creation = ${toLua muc.restrictRoomCreation}
-            max_history_messages = ${toLua muc.maxHistoryMessages}
-            muc_room_locking = ${toLua muc.roomLocking}
-            muc_room_lock_timeout = ${toLua muc.roomLockTimeout}
-            muc_tombstones = ${toLua muc.tombstones}
-            muc_tombstone_expiry = ${toLua muc.tombstoneExpiry}
-            muc_room_default_public = ${toLua muc.roomDefaultPublic}
-            muc_room_default_members_only = ${toLua muc.roomDefaultMembersOnly}
-            muc_room_default_moderated = ${toLua muc.roomDefaultModerated}
-            muc_room_default_public_jids = ${toLua muc.roomDefaultPublicJids}
-            muc_room_default_change_subject = ${toLua muc.roomDefaultChangeSubject}
-            muc_room_default_history_length = ${toLua muc.roomDefaultHistoryLength}
-            muc_room_default_language = ${toLua muc.roomDefaultLanguage}
-            ${ muc.extraConfig }
-        '') cfg.muc}
-
-      ${ lib.optionalString (cfg.uploadHttp != null) ''
-        -- TODO: think about migrating this to mod-http_file_share instead.
-        Component ${toLua cfg.uploadHttp.domain} "http_upload"
-            http_upload_file_size_limit = ${cfg.uploadHttp.uploadFileSizeLimit}
-            http_upload_expire_after = ${cfg.uploadHttp.uploadExpireAfter}
-            ${lib.optionalString (cfg.uploadHttp.userQuota != null) "http_upload_quota = ${toLua cfg.uploadHttp.userQuota}"}
-            http_upload_path = ${toLua cfg.uploadHttp.httpUploadPath}
-      ''}
-
-      ${ lib.concatStringsSep "\n" (lib.mapAttrsToList (n: v: ''
-        VirtualHost "${v.domain}"
-          enabled = ${boolToString v.enabled};
-          ${ optionalString (v.ssl != null) (createSSLOptsStr v.ssl) }
-          ${ v.extraConfig }
-        '') cfg.virtualHosts) }
+    environment.etc."prosody/prosody.cfg.lua".text = pkgs.runCommandLocal "prosody.cfg.lua" {
+      nativeBuildInputs = [ cfg.package ];
+    } ''
+      ln -s ${configFile} $out
+      prosodyctl check config $out
```

This currently runs unconditionally and the `checkConfig` option is unused.
```suggestion
    '' + lib.optionalString cfg.checkConfig ''
      prosodyctl check config $out
```

---

You don't know any tool by chance where I can pick commits from feature branches into a combined branch? I continued testing this on my combined branch and forgot to move some final fixes over. 🙄

---

You could try to move to some other workflow like [git-branchless](https://github.com/arxanas/git-branchless)?